### PR TITLE
scripts to gather dependencies

### PR DESCRIPTION
**Pull request by techcobweb**
Signed-off-by: Mike Cobbett <77053+techcobweb@users.noreply.github.com>

# Why ?
I wanted a fast way of telling me what all our dependencies are for this repository.

Ideally this would be run in every build and the output saved as a build artifact, but I've not done this yet.

### CODE REVIEW

File: tools/list-dependencies.sh [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+#
+# Copyright contributors to the Galasa project
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+
+#-----------------------------------------------------------------------------------------                   
+#
+# Objectives: Build this repository code locally.
```
Comment (jadecarino) [inline tools/list-dependencies.sh]: Do you want to update this for this script to avoid confusion

File: tools/list-dependencies.sh [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+#-----------------------------------------------------------------------------------------                   
+function usage {
+    info "Syntax: list-dependencies.sh [OPTIONS]"
+    cat << EOF
+Lists all the dependencies in all the code in this repository.
+
+Options are:
+-h | --help : Display this help text
+--module The name of the module to start building from
+--chain true/false/yes/no/y/n
```
Comment (jadecarino) [inline tools/list-dependencies.sh]: Are these accepted flags or meant to be removed? I can't see where they are being processed

**Review by jadecarino — COMMENTED** [review]
Approved just a couple of comments

File: tools/list-dependencies.sh (hunk 1 above)
Comment (techcobweb) [reply tools/list-dependencies.sh]: fixed

**Review by jadecarino — APPROVED** [review]

**Pull request merged**